server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -55,9 +55,16 @@ public class DateFormatters {
      * So provide a JVM option so that users can just use the java.time parsers, if they really need to.
      */
     @UpdateForV9    // evaluate if we need to deprecate/remove this
-    private static final boolean JAVA_TIME_PARSERS_ONLY = Booleans.parseBoolean(System.getProperty("es.datetime.java_time_parsers"), false);
+    private static final boolean JAVA_TIME_PARSERS_ONLY;
 
     static {
+        // if this is running in tests, use the seed to determine whether we use the ISO or java.time implementations
+        // to ensure both implementations are tested properly
+        String lastSeedDigit = System.getProperty("tests.seed", "0");
```

Comment:
I would like to avoid production code looking at the test seed. Instead, can we decide whether to set the sysprop in gradle? Or alternatively, set it in BootstrapForTesting (though that would rely on this class not yet being statically initialized).

Reply:
Oh yes, didn't spot that

Reply:
See [37abfcc](https://github.com/elastic/elasticsearch/pull/108944/commits/37abfcc03f222551655fce6f84268b02d2416e93)